### PR TITLE
add checks/logs on broker connect timeout

### PR DESCRIPTION
**Pull request by mattkasun**
using code suggested by @whites11 in PR #1071

### CODE REVIEW

File: netclient/functions/daemon.go [hunk 1]
```diff
@@ -242,6 +242,14 @@ func setupMQTT(cfg *config.ClientConfig, publish bool) mqtt.Client {
 		} else {
 			err = token.Error()
 		}
+		brokerErr := checkBroker(cfg.Server.Server)
+		if brokerErr.Error() == "dns" {
```
Comment (0xdcarns) [inline netclient/functions/daemon.go:246]: this code block 246-252 is repeated below, so probably want to convert to a private func and replace it. You also don't check if that err is `nil` like you do below, which could cause a panic.

File: netclient/functions/mqpublish.go [hunk 2]
```diff
@@ -140,7 +142,16 @@ func publish(nodeCfg *config.ClientConfig, dest string, msg []byte, qos byte) er
 			err = token.Error()
 		}
 		if err != nil {
-			return token.Error()
+			if brokerErr := checkBroker(nodeCfg.Server.Server); brokerErr != nil {
+				if brokerErr.Error() == "dns" {
```
Comment (0xdcarns) [inline netclient/functions/mqpublish.go:146]: see above, turn lines 146-152 into private func.

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: netclient/functions/daemon.go (hunk 1 above)
Comment (mattkasun) [reply netclient/functions/daemon.go]: moved logging to checkBroker func

File: netclient/functions/mqpublish.go (hunk 2 above)
Comment (mattkasun) [reply netclient/functions/mqpublish.go]: moved logging to checkBroker func

**Review by 0xdcarns — APPROVED** [review]

**Pull request merged**